File: tradehistory/source/modules/strategy/src/test/sample_data/inputs/send_message.rb
include_class "org.marketcetera.strategy.ruby.Strategy"
include_class "org.marketcetera.quickfix.FIXVersion"
include_class "org.marketcetera.trade.DestinationID"
include_class "java.lang.Long"
include_class "java.util.Date"
include_class "quickfix.field.TransactTime"

class SendMessage < Strategy
    def on_start
        messageDate = Long.parseLong(get_parameter("date"))

        nullMessage = get_parameter("nullMessage")
        if(nullMessage == nil)
            message = FIXVersion.getFIXVersion("FIX.0.0").getMessageFactory().newBasicOrder()
            message.setField(TransactTime.new(Date.new(messageDate)))
        else
            message = nil
        end

        nullDestination = get_parameter("nullDestination")
        if(nullDestination == nil)
            destination = DestinationID.new("some-destination")
        else
            destination = nil
        end
        
        send_message(message, destination)
    end  
end

File: custom-mdata/source/modules/strategy/src/test/sample_data/inputs/SendMessage.java
import java.util.Date;

import org.marketcetera.quickfix.FIXVersion;
import org.marketcetera.strategy.java.Strategy;
import org.marketcetera.trade.BrokerID;

import quickfix.Message;
import quickfix.field.TransactTime;

/**
 * Test strategy to send messages via the FIX escape hatch.
 *
 * @author <a href="mailto:colin@marketcetera.com">Colin DuPlantis</a>
 * @version $Id$
 * @since 1.0.0
 */
public class SendMessage
        extends Strategy
{
    /* (non-Javadoc)
     * @see org.marketcetera.strategy.java.Strategy#onStart()
     */
    @Override
    public void onStart()
    {
        long messageDate = Long.parseLong(getParameter("date"));
        String nullMessage = getParameter("nullMessage");
        Message message;
        if(nullMessage == null) {
            message = FIXVersion.FIX_SYSTEM.getMessageFactory().newBasicOrder();
            message.setField(new TransactTime(new Date(messageDate)));
        } else {
            message = null;
        }
        String nullBroker = getParameter("nullBroker");
        BrokerID broker;
        if(nullBroker == null) {
            broker = new BrokerID("some-broker");
        } else {
            broker = null;
        }
        sendMessage(message,
                    broker);
        setProperty("onStart",
                    Long.toString(System.currentTimeMillis()));
    }
}
